docs/proposals/security-context-constraints.md

```diff
@@ -0,0 +1,292 @@
+instance, creating a new user within a namespace should not require the cluster administrator to
+define the user's security context constraints.  They should receive the default set of constraints
+that the administrator has defined for users.
+
+When a service account or user is removed finalizers should ensure that the references from the security context
+restraints are also removed in order to avoid the risk of having the same user or service account
+receive permissions inadvertently.  This can also be avoided by using a unique identifier like
+the object UID in the security context constraints list of allowed users but clean up should
+still occur to avoid unnecessary object growth.
+
+#### Default Security Context Constraints And Overrides
+
+In order to establish security context constraints for service accounts and users there must be a way
+to identify the default set of constraints that is to be used.  In order to identify the constraints
+a type field or default flag may be used on the security context constraints object.
+
+To avoid unnecessarily large objects with references to many users in the system the constraints
+identified as the default for the object type can be considered referenceable by all objects
+of that type.  For example, rather than listing every user in the cluster in the default user
+security context constraint object the mechanism that retrieves the available security context
+constraints for the user may fetch all constraints with specific references to the user and
+the default constraints.
+
+If an administrator would like to provide a user with a changed set of security context permissions
+they may do the following:
+
+1.  Create a new security context constraints object and add a reference to the user.
+1.  Create a new security context constraints object and add a reference to a serivce account
+that the user has access to reference.
+
+#### Admission
+
+Admission control using an authorizer allows the ability to control the creation of resources
+based on capabilities granted to a user.  In terms of the security context constraints it means
+that an admission controller may inspect the user info made available in the context to retrieve
+and appropriate set of security context constraints for validation.
+
+The appropriate set of security context constraints is defined as the set of security context
+constraints that the user has references to the user, the default set of constraints for the
+user type, and any security context constraints that they are allowed to reference via a service
+account.
+
+Admission will use the SecurityContextConstraintsAllocator to ensure that any requests for a
+specific security context constraint setting or to generate settings using the following approach:
+
+**Pod with referenced service account**
+
+1.  Ensure that the user has permissions to reference the service account.
```

Comment:
I don't think this is the job of the security context constraint admission control...